### DISCUSSION TITLE
https://github.com/encode/starlette/blob/master/starlette/responses.py - appears UTF incompatible (Line 58 and probably other places)

### DISCUSSION THREAD

**Discussion by mxav1111** [encode/starlette#2579] [Potential Issue]
Hi team
I searched and couldn't find any open issue around this.

Encountered an issue while working on small project that involves UTF-8 double byte character set. In this specific case, it happens to be  Hindi  (Indian Language).

When filename is in hindi language , such as :- ब्रह्मवर्चसकीध्यानधारणा.txt, following error appears in log : -

=====
  File "/home/pta/.pyenv/versions/3.12.2/envs/pyenv_flet1/lib/python3.12/site-packages/starlette/responses.py", line 297, in __init__
    self.init_headers(headers)
  File "/home/pta/.pyenv/versions/3.12.2/envs/pyenv_flet1/lib/python3.12/site-packages/starlette/responses.py", line 58, in init_headers
    (k.lower().encode("latin-1"), v.encode("latin-1"))
                                  ^^^^^^^^^^^^^^^^^^^
=====

Changed responses.py to : -
      k.lower().encode("utf-8"), v.encode("utf-8"))

Thereafter Error went away and file was downloaded with no issues any more.

**Comment by GdeGordon** [comment]
We have encountered the same issue.
Is there any reason why headers are encoded in latin-1? If not I can submit a PR
